AWS Third Party Apache Dependency Failing To Attach to jar file in lambda. Throwing ClassNotFoundException

I'm trying to create a lambda that can interact with Kinesis, Dynamo, or S3. For some reason my lambda cannot communicate with any of these services because an error is being thrown : 
![image](https://user-images.githubusercontent.com/38298138/38639327-cd23c5a6-3d9e-11e8-8a2c-ccec77ad1063.png)
(Full log further down.)

I believe I've created the lambda's jar correctly using mvn package with the maven shade plugin, even going as far as including the entire AWS SDK in my POM in an attempt to get this jar to work.

Seems like what is making it fail is an Apache third party logger used by AWS with their communication methods. Whenever I try to create a Dynamoor S3 client, that is what is triggering the error.

If I were to only write a basic lambda to capture event data from a JSON file and then print it to a log file it works, but once I try intercommunication it fails. 

I'm using Windows 10 with docker-compose up to run localstack. So I don't know how to go about trying to include this dependency in the localstack environment. I was hoping to find a way to include this dependency in my POM and then have it in my jar, but no luck. 

Any advice or insight would be appreciated. 

Here's the simple lambda I was making, just something to delete an object in S3 : https://pastebin.com/8Ah2n3nZ

FULL TRACE : 
aws lambda --endpoint-url=http://localhost:4574  --region us-east-1 invoke --function-name dynamo
Function2 --payload file://input.json dynamoResult.log

An error occurred (InternalFailure) when calling the Invoke operation (reached max retries: 4): Error executing Lambda function: Lambda process returned error status code: 1. Output:
Exception in thread "main" java.lang.ExceptionInInitializerError
        at com.amazonaws.client.builder.AwsClientBuilder.<clinit>(AwsClientBuilder.java:58)
        at com.amazonaws.HelloWorld.handleRequest(HelloWorld.java:76)
        at com.amazonaws.HelloWorld.handleRequest(HelloWorld.java:26)
        at cloud.localstack.LambdaExecutor.main(LambdaExecutor.java:100)
Caused by: com.amazonaws.thirdparty.apache.codec.logging.LogConfigurationException: java.lang.ClassNotFoundException: com.amazonaws.thirdparty.apache.codec.logging.impl.LogFactoryImpl (Caused by java.lang.ClassNotFoundException: com.amazonaws.thirdparty.apache.codec.logging.impl.LogFactoryImpl)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory.createFactory(LogFactory.java:1201)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory$2.run(LogFactory.java:1003)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory.newFactory(LogFactory.java:1000)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory.getFactory(LogFactory.java:626)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory.getLog(LogFactory.java:657)
        at com.amazonaws.regions.AwsRegionProviderChain.<clinit>(AwsRegionProviderChain.java:33)
        ... 4 more
Caused by: java.lang.ClassNotFoundException: com.amazonaws.thirdparty.apache.codec.logging.impl.LogFactoryImpl
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory.createFactory(LogFactory.java:1063)
        ... 10 more
 Traceback (most recent call last):
  File "/opt/code/localstack/localstack/services/awslambda/lambda_api.py", line 255, in run_lambda
    event, context=context, version=version, async=async)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 460, in execute
    process.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 456, in do_execute
    result = lambda_function(event, context)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_api.py", line 388, in execute
    handler=arn_to_lambda[arn].handler, main_file=main_file)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 481, in execute_java_lambda
    result, log_output = self.run_lambda_executor(cmd, async=async)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 67, in run_lambda_executor
    (return_code, log_output))
Exception: Lambda process returned error status code: 1. Output:
Exception in thread "main" java.lang.ExceptionInInitializerError
        at com.amazonaws.client.builder.AwsClientBuilder.<clinit>(AwsClientBuilder.java:58)
        at com.amazonaws.HelloWorld.handleRequest(HelloWorld.java:76)
        at com.amazonaws.HelloWorld.handleRequest(HelloWorld.java:26)
        at cloud.localstack.LambdaExecutor.main(LambdaExecutor.java:100)
Caused by: com.amazonaws.thirdparty.apache.codec.logging.LogConfigurationException: java.lang.ClassNotFoundException: com.amazonaws.thirdparty.apache.codec.logging.impl.LogFactoryImpl (Caused by java.lang.ClassNotFoundException: com.amazonaws.thirdparty.apache.codec.logging.impl.LogFactoryImpl)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory.createFactory(LogFactory.java:1201)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory$2.run(LogFactory.java:1003)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory.newFactory(LogFactory.java:1000)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory.getFactory(LogFactory.java:626)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory.getLog(LogFactory.java:657)
        at com.amazonaws.regions.AwsRegionProviderChain.<clinit>(AwsRegionProviderChain.java:33)
        ... 4 more
Caused by: java.lang.ClassNotFoundException: com.amazonaws.thirdparty.apache.codec.logging.impl.LogFactoryImpl
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at com.amazonaws.thirdparty.apache.codec.logging.LogFactory.createFactory(LogFactory.java:1063)
        ... 10 more

--- a/localstack/ext/java/pom.xml
+++ b/localstack/ext/java/pom.xml
@@ -53,18 +53,6 @@
             <artifactId>winp</artifactId>
             <version>1.23</version>
         </dependency>
-
-        <dependency>
-            <groupId>com.amazonaws</groupId>
-            <artifactId>aws-java-sdk</artifactId>
-            <version>1.11.125</version>
-            <scope>provided</scope>
-        </dependency>
-        <dependency>
-            <groupId>com.amazonaws</groupId>
-            <artifactId>aws-lambda-java-core</artifactId>
-            <version>1.1.0</version>
-        </dependency>
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
@@ -72,26 +60,24 @@
         </dependency>
         <dependency>
             <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk</artifactId>
+            <version>1.11.313</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-lambda-java-core</artifactId>
+            <version>1.2.0</version>
+        </dependency>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
             <artifactId>aws-lambda-java-events</artifactId>
-            <version>1.3.0</version>
-            <exclusions>
-                <exclusion>
-                    <groupId>com.amazonaws</groupId>
-                    <artifactId>aws-java-sdk-s3</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>com.amazonaws</groupId>
-                    <artifactId>aws-java-sdk-sns</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>com.amazonaws</groupId>
-                    <artifactId>aws-java-sdk-kinesis</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>com.amazonaws</groupId>
-                    <artifactId>aws-java-sdk-dynamodb</artifactId>
-                </exclusion>
-            </exclusions>
+            <version>2.1.0</version>
+        </dependency>
+        <dependency>
+            <groupId>commons-logging</groupId>
+            <artifactId>commons-logging</artifactId>
+            <version>1.2</version>
         </dependency>
 
         <!-- test dependencies -->
@@ -120,11 +106,11 @@
         <profile>
             <id>fatjar</id>
             <dependencies>
-                <dependency>
-                    <groupId>com.amazonaws</groupId>
-                    <artifactId>aws-java-sdk</artifactId>
-                    <version>1.11.125</version>
-                </dependency>
+              <dependency>
+                  <groupId>com.amazonaws</groupId>
+                  <artifactId>aws-java-sdk</artifactId>
+                  <version>1.11.313</version>
+              </dependency>
             </dependencies>
         </profile>
     </profiles>
@@ -152,7 +138,7 @@
                     </execution>
                 </executions>
                 <configuration>
-                    <minimizeJar>true</minimizeJar>
+                    <minimizeJar>false</minimizeJar>
                     <createDependencyReducedPom>false</createDependencyReducedPom>
                     <shadedArtifactAttached>true</shadedArtifactAttached>
                     <shadedClassifierName>fat</shadedClassifierName>
@@ -190,19 +176,11 @@
                         </relocation>
                         <relocation>
                             <pattern>org.apache.commons</pattern>
-                            <shadedPattern>com.amazonaws.thirdparty.apache.codec</shadedPattern>
+                            <shadedPattern>com.amazonaws.thirdparty.apache.commons</shadedPattern>
                         </relocation>
                         <relocation>
                             <pattern>software.amazon.ion</pattern>
                             <shadedPattern>com.amazonaws.thirdparty.ion</shadedPattern>
-                        </relocation>
-                        <relocation>
-                            <pattern>org.apache.commons.logging</pattern>
-                            <shadedPattern>com.amazonaws.thirdparty.apache.logging</shadedPattern>
-                        </relocation>
-                        <relocation>
-                            <pattern>org.apache.commons.logging</pattern>
-                            <shadedPattern>com.amazonaws.thirdparty.apache.logging</shadedPattern>
                         </relocation>
                     </relocations>
                 </configuration>

--- a/localstack/ext/java/src/main/java/cloud/localstack/LambdaContext.java
+++ b/localstack/ext/java/src/main/java/cloud/localstack/LambdaContext.java
@@ -1,12 +1,12 @@
 package cloud.localstack;
 
-import java.util.logging.Level;
-import java.util.logging.Logger;
-
 import com.amazonaws.services.lambda.runtime.ClientContext;
 import com.amazonaws.services.lambda.runtime.CognitoIdentity;
 import com.amazonaws.services.lambda.runtime.Context;
 import com.amazonaws.services.lambda.runtime.LambdaLogger;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 public class LambdaContext implements Context {
 
@@ -14,8 +14,15 @@ public class LambdaContext implements Context {
 
 	public LambdaLogger getLogger() {
 		return new LambdaLogger() {
+
+			@Override
 			public void log(String msg) {
 				LOG.log(Level.INFO, msg);
+			}
+
+			@Override
+			public void log(byte[] msg) {
+				log(new String(msg));
 			}
 		};
 	}

